src/main.cpp

```diff
@@ -7507,10 +7569,11 @@ bool SendMessages(const Consensus::Params& params, CNode* pto)
                     if (!txinfo.tx) {
                         continue;
                     }
+                    CInv inv = InvForTransaction(txinfo.tx);
```

Comment:
nit: why compute this before the checks on the next two lines?

Reply:
Partly because IIRC this is where the equivalent change was made in upstream's equivalent PR. Partly because there's a subsequent PR that adds an extra check here.